File: kotlin-examples/src/main/java/Interoperability.java
import java.util.ArrayList;
import java.util.List;

public class Interoperability {

    public static void main(String[] args) {
        Person p = new Person("Martin");
        Person p2 = new Person("Martin", 31);
    }

    public static List<String> mutableList() {
        return new ArrayList<String>();
    }

}

File: kotlin-examples/src/main/kotlin/Interoperability.kt
class InteroperabilityKt {

    fun accessMap() {
        val list = ArrayList<String>()
        val value: String? = list.get(0)
    }

    fun accessMutableList() {
        val mutableList: List<String> = Interoperability.mutableList()
        val mutableList2: MutableList<String> = Interoperability.mutableList()
    }

}